src/presentation/http/router/noteSettings.ts

```diff
@@ -173,6 +175,54 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(team);
   });
 
+  /**
+   * Generates a new invitation hash for a specific note
+   */
+  fastify.patch<{
+    Params: {
+      notePublicId: NotePublicId;
+    },
+    Reply: InvitationHash,
+  }>('/:notePublicId/invitation-hash', {
+    config: {
+      policy: [
+        'authRequired',
+        'userInTeam',
+      ],
+    },
+    schema: {
+      params: {
+        notePublicId: {
+          $ref: 'NoteSchema#/properties/id',
+        },
+      },
+    },
+    preHandler: [
+      noteResolver,
+    ],
+  }, async (request, reply) => {
+    const noteId = request.note?.id as number;
+
+    /**
+     * Generates invitation hash
+     */
+    const invitationHash = createInvitationHash();
```

Comment:
Business logic should be placed in domain service

Reply:
Done